virtualizarr/xarray.py

```diff
@@ -1,16 +1,256 @@
-from collections.abc import Iterable, Mapping
+from __future__ import annotations
+
+import os
+from collections.abc import Callable, Iterable, Mapping, MutableMapping, Sequence
+from concurrent.futures import Executor
+from pathlib import Path
 from typing import (
+    TYPE_CHECKING,
     Any,
     Hashable,
-    MutableMapping,
+    Literal,
     Optional,
+    cast,
 )
 
 import xarray as xr
 import xarray.indexes
+from obstore.store import ObjectStore
+from xarray import DataArray, Dataset, Index, combine_by_coords
+from xarray.backends.common import _find_absolute_paths
+from xarray.core.types import NestedSequence
+from xarray.structure.combine import _infer_concat_order_from_positions, _nested_combine
 
 from virtualizarr.manifests import ManifestStore
-from virtualizarr.utils import _FsspecFSFromFilepath
+from virtualizarr.manifests.manifest import validate_and_normalize_path_to_uri
+from virtualizarr.parallel import get_executor
+from virtualizarr.parsers import Parser
+
+if TYPE_CHECKING:
+    from xarray.core.types import (
+        CombineAttrsOptions,
+        CompatOptions,
+        JoinOptions,
+    )
+
+
+def open_virtual_dataset(
+    file_url: str,
+    object_store: ObjectStore,
+    parser: Parser,
+    drop_variables: Iterable[str] | None = None,
+    loadable_variables: Iterable[str] | None = None,
+    decode_times: bool | None = None,
+    cftime_variables: Iterable[str] | None = None,
+    indexes: Mapping[str, xr.Index] | None = None,
+) -> xr.Dataset:
```

Comment:
This has removed several arguments from the API signature, including `group` and `reader_options`. 

It would be kinder to our users to continue accepting these but throw a deprecation warning. This is a fair bit of work to get right so we could do this in a follow up PR - the important thing is that we do it before release.

Reply:
@TomNicholas I wonder if this would be better captured by documentation do to the fact that we will no longer be controlling parser instantiation and this will now be user managed code.  As an example, I'm building a new HRRR specific parser that will not have group support and it will exist outside of Virtualizarr and the constructor will only ever be called in user space.